src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,282 @@
+        assert os.path.exists(moved_dir) and not os.path.exists(sub_dir)
+
+    def test_dir_has_nestedfd(self, empty_dir, nested_fd):
+        """
+        When the files are nested in subdirectories more than 2 directories deep
+        """
+        dest_dir = empty_dir
+        source_dir = tempfile.mkdtemp()
+        sub_dir1 = tempfile.mkdtemp(dir=source_dir)
+        sub_dir2 = tempfile.mkdtemp(dir=sub_dir1)
+        fd, source_file = tempfile.mkstemp(dir=sub_dir2)
+        move_tree(source_dir, dest_dir)
+        moved_dir1 = os.path.join(dest_dir, os.path.basename(sub_dir1))
+        moved_dir2 = os.path.join(moved_dir1, os.path.basename(sub_dir2))
+        moved_file = os.path.join(moved_dir2, os.path.basename(source_file))
+        assert os.path.exists(moved_dir1) and not os.path.exists(sub_dir1)
+        assert os.path.exists(moved_dir2)
+        assert os.path.exists(moved_file)
+
+
+class TestFdOpen:
+    def test_open_dir(self):
+        with tempfile.TemporaryDirectory() as dir:
+            dir_fd = os.open(dir, os.O_RDONLY)
+            with fd_open(dir) as fdd:
+                same_dir = os.path.sameopenfile(fdd, dir_fd)
+            assert same_dir
+
+    def test_open_file(self):
+        with tempfile.NamedTemporaryFile() as file:
+            file_fd = os.open(file.name, os.O_RDONLY)
+            with fd_open(file.name) as fdf:
+                same_file = os.path.sameopenfile(fdf, file_fd)
+            assert same_file
+
+    def test_close(self):
+        with tempfile.TemporaryDirectory() as dir:
+            with fd_open(dir) as fdd:
+                dir_fd = fdd
+            with pytest.raises(Exception):
+                os.close(dir_fd)
+
+
+def test_copy_test_script_files(dir_has_onefile):
+    markus_address = "http://localhost:3000/csc108/en/main"
+    assignment_id = 1
+    tests_path, test_file = dir_has_onefile
+    copy_test_script_files(markus_address, assignment_id, tests_path)
+    test_script_outer_dir = test_script_directory(
+        markus_address, assignment_id
+    )
+    test_script_dir = os.path.join(test_script_outer_dir, FILES_DIRNAME)
+    copied_file = os.path.join(test_script_dir, test_file)
+    assert os.path.exists(copied_file)
+
+
+def copy_test_script_files(markus_address, assignment_id, tests_path):
+    """
+        Copy test script files for a given assignment to the tests_path
+        directory if they exist. tests_path may already exist and contain
+        files and subdirectories.
+        """
+    test_script_outer_dir = test_script_directory(
+        markus_address, assignment_id
+    )
+    test_script_dir = os.path.join(test_script_outer_dir, FILES_DIRNAME)
+    if os.path.isdir(test_script_dir):
+        with fd_open(test_script_dir) as fd:
+            with fd_lock(fd, exclusive=False):
+                return copy_tree(test_script_dir, tests_path)
+    return []
+
+
+def test_script_directory(markus_address, assignment_id, set_to=None):
```

Comment:
Is this a test or a helper function? Either way, it should be removed because it is overriding the function from redis_management.py